src/scripts/pokemons/PokemonFactory.ts

```diff
@@ -155,4 +157,20 @@ class PokemonFactory {
         const catchRateRaw = Math.floor(Math.pow(baseCatchRate, 0.75)) + catchVariation;
         return GameConstants.clipNumber(catchRateRaw, 0, 100);
     }
+
+    private static generateHeldItem(item: string, chance: number): string | null {
+        if (!item || !ItemList[item]) {
+            return null;
+        }
+
+        if (EffectEngineRunner.isActive(GameConstants.BattleItemType.Item_magnet)()) {
+            chance /= 1.5;
+        }
+        const rand: number = Math.floor(Math.random() * chance) + 1;
+        if (rand <= 1) {
```

Comment:
If we just did without adding the 1:
```ts
        const rand: number = Math.floor(Math.random() * chance) ;
```
Couldn't we just check for:
```ts
        if (!rand) {
```

Reply:
rand should be an integer between 1 and chance, so we could just be checking `rand === 1` instead. I copied the code for shiny generation, we should probably change it there as well.

Reply:
It might even been worth just adding a function to GameHelper or something
Maybe like this:
```ts
chanceSuccess = (totalChance, successChance = 1) => {,
    // Add 1 for min success chance
    const rand = Math.floor(Math.random() * totalChance) + 1;
    return rand <= successChance;
}
```

Reply:
We can do this in a separate PR though.